source/server.go

```diff
@@ -33,18 +33,20 @@ type Server struct {
 
 	RecordCh chan sdk.Record
 
-	teardown    chan struct{}
-	openContext context.Context
+	teardown      chan struct{}
+	streamIsSetCh chan struct{}
```

Comment:
Can we please add a comment on this field explaining how it's used? Like what it means if the channel contains a value vs. if it's empty and why we need it.